nav2_costmap_2d/src/costmap_2d_publisher.cpp

```diff
@@ -170,7 +170,9 @@ void Costmap2DPublisher::publishCostmap()
 {
   if (node_->count_subscribers(costmap_raw_pub_->get_topic_name()) > 0) {
     prepareCostmap();
-    costmap_raw_pub_->publish(costmap_raw_);
+
+    auto costmap_raw_ptr = std::make_unique<nav2_msgs::msg::Costmap>(std::move(costmap_raw_));
```

Comment:
This seems unnecessary. What's the type of costmap_raw_? Can you make it into a unique pointer to begin with in the class variable? Then you just create the unique_ptr in the prepareCostmap and then just std::move to publish it. Right now, you're making 2 copies.

Reply:
Yes, that is better. I should make the similar changes with `prepareGrid()` function as well.